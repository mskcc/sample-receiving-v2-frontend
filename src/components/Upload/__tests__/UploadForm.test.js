import React, { Component } from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import TestUploadForm from '../UploadForm'

// import { initialFormStateMock } from '../../../mocks'

const UploadForm = props => <TestUploadForm {...props} />

describe('Render UploadForm', () => {
  it('render UploadForm correctly', () => {
    const UploadFormComponent = renderer.create(<UploadForm />).toJSON()
    expect(UploadFormComponent).toMatchSnapshot()
  })
})

describe('responds to input', () => {
  it('has material field', () => {
    const wrapper = mount(<UploadForm />)
    expect(wrapper.find('#material').exists()).toBe(true)
  })

  // it('responds to click', () => {
  //   const fakeEvent = { preventDefault: () => console.log('preventDefault') }
  //   const validate = jest.fn()
  //   const wrapper = mount(<UploadForm />)

  //   wrapper.find('#upload-form').simulate('submit', fakeEvent)
  //   expect(validate).toHaveBeenCalled()
  // })
})
