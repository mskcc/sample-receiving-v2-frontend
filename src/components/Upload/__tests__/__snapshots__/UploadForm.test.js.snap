// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UploadForm render UploadForm correctly 1`] = `
<div
  className="MuiPaper-root-9 MuiPaper-elevation1-12 MuiPaper-rounded-10 UploadForm-container-1"
>
  <form
    className="UploadForm-form-2"
    id="upload-form"
    onSubmit={[Function]}
  >
    <div
      className="Dropdown-textField-36"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-0-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-37 MuiFormControl-fullWidth-40"
        >
          <label
            className="MuiFormLabel-root-52 MuiInputLabel-root-41 MuiInputLabel-formControl-46 MuiInputLabel-animated-49"
            data-shrink={false}
            htmlFor="downshift-0-input"
            id="downshift-0-label"
          >
            Missing translationId: upload.form.material_label for language: \${ languageCode }
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-0-label"
            className="MuiInputBase-root-72 MuiInput-root-59 MuiInput-underline-63 MuiInputBase-formControl-73 MuiInput-formControl-60 MuiInputBase-adornedEnd-77"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-82 MuiInput-input-67 MuiInputBase-inputAdornedEnd-88"
              disabled={false}
              id="material"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={false}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-89 MuiInputAdornment-positionEnd-92"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-100 MuiIconButton-root-94"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-99"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-103"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-209"
                />
              </button>
            </div>
          </div>
          <p
            className="MuiFormHelperText-root-112"
          >
            Missing translationId: upload.form.material_helptext for language: \${ languageCode } (1 choices)
          </p>
        </div>
      </div>
    </div>
    <div
      className="Dropdown-textField-36"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-1-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-37 MuiFormControl-fullWidth-40"
        >
          <label
            className="MuiFormLabel-root-52 MuiInputLabel-root-41 MuiInputLabel-formControl-46 MuiInputLabel-animated-49"
            data-shrink={false}
            htmlFor="downshift-1-input"
            id="downshift-1-label"
          >
            Missing translationId: upload.form.application_label for language: \${ languageCode }
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-1-label"
            className="MuiInputBase-root-72 MuiInput-root-59 MuiInput-underline-63 MuiInputBase-formControl-73 MuiInput-formControl-60 MuiInputBase-adornedEnd-77"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-82 MuiInput-input-67 MuiInputBase-inputAdornedEnd-88"
              disabled={false}
              id="application"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={false}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-89 MuiInputAdornment-positionEnd-92"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-100 MuiIconButton-root-94"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-99"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-103"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-209"
                />
              </button>
            </div>
          </div>
          <p
            className="MuiFormHelperText-root-112"
          >
            Missing translationId: upload.form.application_helptext for language: \${ languageCode } (1 choices)
          </p>
        </div>
      </div>
    </div>
    <fieldset
      className="MuiFormControl-root-37"
    >
      <div
        className="Dropdown-textField-36"
      >
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-labelledby="downshift-2-label"
          aria-owns={null}
          role="combobox"
        >
          <div
            className="MuiFormControl-root-37 MuiFormControl-fullWidth-40"
          >
            <label
              className="MuiFormLabel-root-52 MuiInputLabel-root-41 MuiInputLabel-formControl-46 MuiInputLabel-animated-49"
              data-shrink={false}
              htmlFor="downshift-2-input"
              id="downshift-2-label"
            >
              Missing translationId: upload.form.species_label for language: \${ languageCode }
            </label>
            <div
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls={null}
              aria-labelledby="downshift-2-label"
              className="MuiInputBase-root-72 MuiInput-root-59 MuiInput-underline-63 MuiInputBase-formControl-73 MuiInput-formControl-60 MuiInputBase-adornedEnd-77"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="off"
                className="MuiInputBase-input-82 MuiInput-input-67 MuiInputBase-inputAdornedEnd-88"
                disabled={false}
                id="species"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                required={false}
                type="text"
                value=""
              />
              <div
                className="MuiInputAdornment-root-89 MuiInputAdornment-positionEnd-92"
              >
                <button
                  aria-label="Toggle menu open"
                  className="MuiButtonBase-root-100 MuiIconButton-root-94"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label-99"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-103"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-209"
                  />
                </button>
              </div>
            </div>
            <p
              className="MuiFormHelperText-root-112"
            >
              Missing translationId: upload.form.species_helptext for language: \${ languageCode } (1 choices)
            </p>
          </div>
        </div>
      </div>
    </fieldset>
    <div
      className="Dropdown-textField-36"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-3-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-37 MuiFormControl-fullWidth-40"
        >
          <label
            className="MuiFormLabel-root-52 MuiInputLabel-root-41 MuiInputLabel-formControl-46 MuiInputLabel-animated-49"
            data-shrink={false}
            htmlFor="downshift-3-input"
            id="downshift-3-label"
          >
            Missing translationId: upload.form.container_label for language: \${ languageCode }
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-3-label"
            className="MuiInputBase-root-72 MuiInput-root-59 MuiInput-underline-63 MuiInputBase-formControl-73 MuiInput-formControl-60 MuiInputBase-adornedEnd-77"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-82 MuiInput-input-67 MuiInputBase-inputAdornedEnd-88"
              disabled={false}
              id="container"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={false}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-89 MuiInputAdornment-positionEnd-92"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-100 MuiIconButton-root-94"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-99"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-103"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-209"
                />
              </button>
            </div>
          </div>
          <p
            className="MuiFormHelperText-root-112"
          >
            Missing translationId: upload.form.container_helptext for language: \${ languageCode }
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root-37 Input-textField-121"
    >
      <label
        className="MuiFormLabel-root-52 MuiInputLabel-root-41 MuiInputLabel-formControl-46 MuiInputLabel-animated-49"
        data-shrink={false}
        htmlFor="number_of_samples"
      >
        Missing translationId: upload.form.number_of_samples_label for language: \${ languageCode }
      </label>
      <div
        className="MuiInputBase-root-72 MuiInput-root-59 MuiInput-underline-63 MuiInputBase-formControl-73 MuiInput-formControl-60"
        onClick={[Function]}
      >
        <input
          aria-describedby="number_of_samples-helper-text"
          aria-invalid={false}
          className="MuiInputBase-input-82 MuiInput-input-67 MuiInputBase-inputType-85 MuiInput-inputType-70"
          disabled={false}
          id="number_of_samples"
          min={0}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="number"
        />
      </div>
      <p
        className="MuiFormHelperText-root-112"
        id="number_of_samples-helper-text"
      >
        Missing translationId: upload.form.number_of_samples_helptext for language: \${ languageCode }
      </p>
    </div>
    <fieldset
      className="MuiFormControl-root-37"
    >
      <div
        className="MuiFormControl-root-37 Input-textField-121"
      >
        <label
          className="MuiFormLabel-root-52 MuiInputLabel-root-41 MuiInputLabel-formControl-46 MuiInputLabel-animated-49 MuiInputLabel-shrink-48"
          data-shrink={true}
          htmlFor="igo_request_id"
        >
          Missing translationId: upload.form.igo_request_id_label for language: \${ languageCode }
        </label>
        <div
          className="MuiInputBase-root-72 MuiInput-root-59 MuiInput-underline-63 MuiInputBase-formControl-73 MuiInput-formControl-60 MuiInputBase-adornedStart-76"
          onClick={[Function]}
        >
          <div
            className="MuiInputAdornment-root-89 MuiInputAdornment-positionStart-91"
          >
            <p
              className="MuiTypography-root-122 MuiTypography-body1-131 MuiTypography-colorTextSecondary-155"
            >
              IGO-
            </p>
          </div>
          <input
            aria-describedby="igo_request_id-helper-text"
            aria-invalid={false}
            className="MuiInputBase-input-82 MuiInput-input-67 MuiInputBase-inputType-85 MuiInput-inputType-70 MuiInputBase-inputAdornedStart-87"
            disabled={false}
            id="igo_request_id"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="number"
            value=""
          />
        </div>
        <p
          className="MuiFormHelperText-root-112"
          id="igo_request_id-helper-text"
        >
          Missing translationId: upload.form.igo_request_id_helptext for language: \${ languageCode }
        </p>
      </div>
      <div
        className="CheckboxComponent-container-158"
      >
        <label
          className="MuiFormControlLabel-root-161"
        >
          <span
            className="MuiButtonBase-root-100 MuiIconButton-root-94 MuiPrivateSwitchBase-root-173 MuiCheckbox-root-167 MuiCheckbox-colorSecondary-172"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-99"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-103 MuiSvgIcon-fontSizeSmall-110"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
              <input
                checked={false}
                className="MuiPrivateSwitchBase-input-176"
                data-indeterminate={false}
                disabled={false}
                onChange={[Function]}
                type="checkbox"
                value="igo_alternative_id"
              />
            </span>
            <span
              className="MuiTouchRipple-root-209"
            />
          </span>
          <span
            className="MuiTypography-root-122 MuiTypography-body1-131 MuiFormControlLabel-label-166"
          >
            Missing translationId: upload.form.igo_alternative_id_label for language: \${ languageCode }
          </span>
        </label>
        <div>
          <div
            className="MuiPaper-root-9 MuiPaper-elevation1-12 MuiPaper-rounded-10 MuiCard-root-177 CheckboxComponent-card-159"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <div
              className="MuiCardContent-root-178 CheckboxComponent-content-160"
            >
              <p
                className="MuiTypography-root-122 MuiTypography-body1-131 MuiTypography-colorTextSecondary-155"
              >
                Missing translationId: upload.form.igo_alternative_id_helptext for language: \${ languageCode }
                 
              </p>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
  </form>
  <div
    className="Button-wrapper-180"
  >
    <button
      className="MuiButtonBase-root-100 MuiButton-root-183 MuiButton-contained-194 MuiButton-containedSecondary-196 MuiButton-raised-197 MuiButton-raisedSecondary-199 Button-button-179"
      disabled={false}
      form="upload-form"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="submit"
    >
      <span
        className="MuiButton-label-184"
      >
        Missing translationId: upload.form_submit_label for language: \${ languageCode }
      </span>
      <span
        className="MuiTouchRipple-root-209"
      />
    </button>
    <div
      className="Button-nothingToSubmit-182"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      Missing translationId: upload.form.nothing_to_change for language: \${ languageCode }
    </div>
  </div>
   
</div>
`;
